homework/index.js

```diff
@@ -30,18 +32,89 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
+  function main() {
+    fetchJSON(HYF_REPOS_URL, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        createAndAppend('div', root, {
+          text: err.message,
+          class: 'alert-error',
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const jsonString = JSON.stringify(data, null, 2);
+        const div = createAndAppend('div', root, {
+          id: 'repoDescriptionDiv',
+        });
+        createAndAppend('h2', div, { text: 'HYF Repositories' });
+        const select = createAndAppend('select', div, {
+          id: 'selectionTree',
+        });
+        createAndAppend('p', div, {
+          id: 'repoDescriptionP',
+        });
+        let valueNumber = 0;
+
+        const repositories = JSON.parse(jsonString, null);
+
+        let contributorsURL = '';
+
+        repositories.forEach(repo => {
+          const options = createAndAppend('option', select, {
+            value: valueNumber,
+          });
+          options.textContent = repo.name;
+          valueNumber += 1;
+        });
+
+        const selectElement = document.getElementById('selectionTree');
+        selectElement.addEventListener('change', event => {
```

Comment:
On every change event you are appending the info about the repoContributors but not removing the info about the previous selected repo so the contributors get added to a long list, not from the repo selected. Try to write some code to remove the repoContributors from the previous one and only show the repoContributors from the selected repo!

Reply:
I've fixed this issue on week2 update. Added a function that deletes child element before adding the new contributor div.